fix naming to match litmus-2.1.3 helm chart in ingress doc (#137)
litmuschaos/litmus-docs: docs/coredns-pod-delete.md

---
id: coredns-pod-delete
title: CoreDNS Pod Delete Experiment Details
sidebar_label: CoreDNS Pod Delete
---
------

## Experiment Metadata

| Type      | Description              | Tested K8s Platform                                               |
| ----------| ------------------------ | ------------------------------------------------------------------|

> | CoreDNS   | CoreDNS pod delete experiment | KubeADM, Minikube  |

Reviewer: ```suggestion
| CoreDNS   | CoreDNS pod delete experiment | Kubeadm, Minikube  |
```

## Prerequisites

- Ensure that the Litmus Chaos Operator is running

> - Ensure that the `coredns-pod-delete` experiment resource is available in the cluster. If not, install from [here](https://hub.litmuschaos.io/api/chaos?file=charts/coredns/coredns-pod-delete/experiment.yaml)

Reviewer: lets add command to check (refer: drain node)

## Entry Criteria

- CoreDNS pods are healthy before chaos injection

> - Service resolution checks by deploying a dummy nginx application

Reviewer: ```suggestion
- Service resolution works successfully as determined by deploying a sample nginx application
```

## Exit Criteria

- CoreDNS pods are healthy post chaos injection

> - Service resolution checks by deploying a dummy nginx application

Reviewer: we can add above line and also about the liveness app

## Details

- Causes graceful pod failure of an coreDNS application
- Tests deployment sanity (replica availability & uninterrupted service) and recovery workflow of the service

> - Deletion of CoreDNS pod will failed the kubernetes service resolution.

Reviewer: do we need this? if the CoreDNS deployment is not resilient (say, not enough replicas), then service resolution will be failed.

## Integrations

- Pod failures can be effected using one of these chaos libraries: `litmus`

## Steps to Execute the Chaos Experiment

- This Chaos Experiment can be triggered by creating a ChaosEngine resource on the cluster. To understand the values to provide in a ChaosEngine specification, refer [Getting Started](getstarted.md/#prepare-chaosengine)

- Follow the steps in the sections below to prepare the ChaosEngine & execute the experiment.

### Prepare ChaosEngine

> - Provide the application info in `spec.appinfo`

Reviewer: comment - about coreDNS ns (kube-system), labels

- Override the experiment tunables if desired

#### Supported Experiment Tunables

| Variables             | Description                                                  | Type      | Notes                                                      |
| ----------------------| ------------------------------------------------------------ |-----------|------------------------------------------------------------|
| TOTAL_CHAOS_DURATION  | The time duration for chaos insertion (seconds)              | Optional  | Defaults to 15s                                            |
| CHAOS_INTERVAL        | Time interval b/w two successive pod failures (sec)          | Optional  | Defaults to 5s                                             |
| LIB                   | The chaos lib used to inject the chaos                       | Optional  | Defaults to `litmus`. Supported: `litmus`                  |

#### Sample ChaosEngine Manifest

```yaml
apiVersion: litmuschaos.io/v1alpha1
kind: ChaosEngine
metadata:

>   name: engine-nginx

Reviewer: engine-coredns

    applabel: 'k8s-app=kube-dns'
    appkind: deployment
  chaosType: 'infra'    # It can be app/infra
  chaosServiceAccount: nginx
  monitoring: false
  jobCleanUpPolicy: delete  # It can be delete/retain
  experiments:
    - name: coredns-pod-delete
      spec:
        components:
           # set chaos duration (in sec) as desired
          - name: TOTAL_CHAOS_DURATION
            value: '30'
            
          # set chaos interval (in sec) as desired
          - name: CHAOS_INTERVAL
            value: '10'
            
          - name: APP_NAMESPACE
            value: 'kube-system'

          # provide application labels
          - name: APP_LABEL
            value: 'k8s-app=kube-dns'

           # provide application kind
          - name: APP_KIND
            value: 'deployment'
            
          - name: CHAOS_NAMESPACE
            value: 'kube-system'
```

### Create the ChaosEngine Resource

- Create the ChaosEngine manifest prepared in the previous step to trigger the Chaos.

  `kubectl apply -f chaosengine.yml`

### Watch Chaos progress

- View coredns pod terminations & recovery by setting up a watch on the coredns pods in the application namespace

>   `watch -n 1 kubectl get pods -n <chaos-namespace>`

Reviewer: may also want to check coreDNS pods health in kube-system

---
id: coredns-pod-delete
title: CoreDNS Pod Delete Experiment Details
sidebar_label: CoreDNS Pod Delete
---
------

## Experiment Metadata

| Type      | Description                   | Tested K8s Platform    |
| ----------| ------------------------------| ---------------------- |
| CoreDNS   | CoreDNS pod delete experiment | Kubeadm, Minikube      |

## Prerequisites

> - Ensure that Litmus is install. If not, install from [here](https://docs.litmuschaos.io/docs/getstarted/)

Reviewer 2: ```Ensure that the Litmus Chaos Operator is running by executing `kubectl get pods` in operator namespace (typically, `litmus`). If not, install from [here](https://github.com/litmuschaos/chaos-operator/blob/master/deploy/operator.yaml)```
Use this line @rajdas98

- Ensure that the `coredns-pod-delete` experiment resource is available in the cluster by executing `kubectl get chaosexperiments` in the desired namespace. If not, install from [here](https://hub.litmuschaos.io/api/chaos?file=charts/coredns/coredns-pod-delete/experiment.yaml)

## Entry Criteria

- CoreDNS replicas are healthy before chaos injection
- Service resolution works successfully as determined by deploying a sample nginx application and a custom liveness app querying the nginx application

## Exit Criteria

- CoreDNS replicas are healthy after chaos injection
- Service resolution works successfully as determined by deploying a sample nginx application and a custom liveness app querying the nginx application

## Details

- Causes graceful pod failure of an coreDNS replicas
- Tests deployment sanity (replica availability & uninterrupted service) and recovery workflow of the service

> - Service resolution will failed if coredns replicas will not present.

Reviewer: Will fail if coredns replicas are not present

## Integrations

- Pod failures can be effected using one of these chaos libraries: `litmus`

## Steps to Execute the Chaos Experiment

- This Chaos Experiment can be triggered by creating a ChaosEngine resource on the cluster. To understand the values to provide in a ChaosEngine specification, refer [Getting Started](getstarted.md/#prepare-chaosengine)

> - Follow the steps in the sections below to prepare the ChaosEngine & execute the experiment.

Reviewer: There is a slightly different line we are using here - pls refer other experiments

---
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: ClusterRoleBinding
metadata:
  name: coredns-sa
  labels:
    name: coredns-sa
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: coredns-sa
subjects:
- kind: ServiceAccount
  name: coredns-sa
  namespace: kube-system
```

### Prepare ChaosEngine

- Provide the application info in `spec.appinfo`
  - It will be default as
    ```
      appinfo:
        appns: kube-system
        applabel: 'k8s-app=kube-dns'
        appkind: deployment
    ```

- Override the experiment tunables if desired

#### Supported Experiment Tunables

| Variables             | Description                                         | Type      | Notes           |
| ----------------------|-----------------------------------------------------|-----------|-----------------|
| TOTAL_CHAOS_DURATION  | The time duration for chaos insertion (seconds)     | Mandatory | Defaults to 15s |
| CHAOS_INTERVAL        | Time interval b/w two successive pod failures (sec) | Mandatory | Defaults to 5s | LIB                   | The chaos lib used to inject the chaos                | Optional  | Defaults to `litmus`, Supported: `litmus`  |

> 

Reviewer 3: can we have the list of all supported variables here like what you have mentioned in the example of chaosengine yaml

#### Sample ChaosEngine Manifest
```yaml
apiVersion: litmuschaos.io/v1alpha1
kind: ChaosEngine
metadata:
  name: engine-coredns
  namespace: kube-system
spec:
  appinfo:
    appns: kube-system
    applabel: 'k8s-app=kube-dns'
    appkind: deployment

>   chaosType: 'infra'    # It can be infra only

Reviewer 2: comment should come above the line

  components:
    runner:

>        image: litmuschaos/ansible-runner:ci

Reviewer 2: ```components:
    runner:
      image: "litmuschaos/chaos-executor:1.0.0"
      type: "go"``` 
Use this here for go executer

---
id: coredns-pod-delete
title: CoreDNS Pod Delete Experiment Details
sidebar_label: CoreDNS Pod Delete
---
------

## Experiment Metadata

| Type      | Description                   | Tested K8s Platform    |
| ----------| ------------------------------| ---------------------- |
| CoreDNS   | CoreDNS pod delete experiment | Kubeadm, Minikube      |

## Prerequisites

> - Ensure that the Litmus Chaos Operator is running by executing `kubectl get pods` in operator namespace (typically, `litmus`). If not, install from [here](https://github.com/litmuschaos/chaos-operator/blob/master/deploy/operator.yaml)

Reviewer 3: Update the operator link with latest from litmuschaos/pages

  resources: ["services", "pods","jobs","chaosengines","chaosexperiments","chaosresults"]
  verbs: ["create","list","get","patch","delete"]
---
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: ClusterRoleBinding
metadata:
  name: coredns-sa
  labels:
    name: coredns-sa
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: coredns-sa
subjects:
- kind: ServiceAccount
  name: coredns-sa
  namespace: kube-system
```

### Prepare ChaosEngine

- Provide the application info in `spec.appinfo`
  - It will be default as
    ```
      appinfo:
        appns: kube-system
        applabel: 'k8s-app=kube-dns'
        appkind: deployment
    ```

- Override the experiment tunables if desired

#### Supported Experiment Tunables

| Variables             | Description                                         | Type      | Notes           |
| ----------------------|-----------------------------------------------------|-----------|-----------------|
| TOTAL_CHAOS_DURATION  | The time duration for chaos insertion (seconds)     | Mandatory | Defaults to 15s |
| CHAOS_INTERVAL        | Time interval b/w two successive pod failures (sec) | Mandatory | Defaults to 5s  | 

> | APP_NAMESPACE         | Namespace in which application pods are deployed	  | Mandatory | Defaults to kube-system |

Reviewer 3: ```suggestion
| APP_NAMESPACE         | Namespace in which application pods are deployed	  | Mandatory | Defaults to `kube-system` |
```

> | APP_LABEL             | Unique Labels in `key=value` format of application deployment	 | Mandatory | Defaults to k8s-app=kube-dns |

Reviewer 3: ```suggestion
| APP_LABEL             | Unique Labels in `key=value` format of application deployment	 | Mandatory | Defaults to `k8s-app=kube-dns` |
```

> | APP_KIND              | This is describe the application Kind | Mandatory | Mandatory | Defaults to Deployment | 

Reviewer 3: ```suggestion
| APP_KIND              | This is describe the application Kind | Mandatory | Mandatory | Defaults to `Deployment` | 
```

> | CHAOS_NAMESPACE       | This is a chaos namespace which will create all infra chaos resources in that namespace	| Mandatory | Default to kube-system |

Reviewer 3: ```suggestion
| CHAOS_NAMESPACE       | This is a chaos namespace which will create all infra chaos resources in that namespace	| Mandatory | Default to `kube-system` |
```
